test/command_runner_test.dart

```diff
@@ -50,6 +50,30 @@ Global options:
 Available commands:
   foo   Set a value.
 
+Run "test help <command>" for more information about a command.'''));
+    });
+
+    test('displays categories', () {
+      runner.addCommand(Category1Command());
+      runner.addCommand(Category2Command());
+      runner.addCommand(FooCommand());
+
+      expect(runner.usage, equals('''
+A test command runner.
+
+Usage: test <command> [arguments]
+
+Global options:
+-h, --help    Print this usage information.
+
+Available commands:
+  foo   Set a value.
+Displayers
```

Comment:
Is it intentional that there is no extra newline above this but there is an extra newline between the categories?

Reply:
Yes, but it's a style choice I'll leave to you. Some other options:

Newlines 
```
Available commands:

Displayers
  baz   Display a value.

Printers
  bar   Print a value.
```

```
Available commands:
  foo   Set a value.

Displayers
  baz   Display a value.

Printers
  bar   Print a value.
```

Newlines + indent categories
```
Available commands:
  
  Displayers
  baz   Display a value.
  
  Printers
  bar   Print a value.
```

```
Available commands:
  foo   Set a value.
  
  Displayers
  baz   Display a value.
  
  Printers
  bar   Print a value.
```

No newlines 
```
Available commands:
Displayers
  baz   Display a value.
Printers
  bar   Print a value.
```

```
Available commands:
  foo   Set a value.
Displayers
  baz   Display a value.
Printers
  bar   Print a value.
```

Reply:
I would personally do either no newlines or newlines always. In general I think the current style is pretty reasonable, but I do think a newline separating the categories from the uncategorized section would be good :).

I am not sure who all to seek opinion from here so ill just cc a few that I know 🤣  @natebosch @munificent @jonahwilliams

Reply:
I don't have any strong opinions here, though big +1 on this feature

Reply:
Thanks! I went with always newlines